packages/k8s.contrib.crd/internal/ModuleGenerator.pkl

```diff
@@ -0,0 +1,366 @@
+moduleTypes: Type.TypeNames = convertedSchemas.map((schema, importAndType) -> Pair(schema, importAndType.type)) + classSchemas
+
+/// Generate a Pkl [ModuleNode] from a given schema.
+moduleNode: ModuleNode =
+  let (allTypeNames = moduleTypes)
+    new {
+      imports {
+        new {
+          value = "package://pkg.pkl-lang.org/pkl-k8s/k8s@1.0.1#/apimachinery/pkg/apis/meta/v1/ObjectMeta.pkl"
+        }
+        for (_import in convertedSchemas.values.map((it) -> it._import).toSet()) {
+          new {
+            value = _import
+          }
+        }
+      }
+      classes {
+        for (schema, type in classSchemas) {
+          generatePklClass(schema, type, allTypeNames)
+        }
+      }
+      declaration {
+        docComment = getDocComment(filteredRootSchema, "module")
+        moduleHeader {
+          name {
+            parts {
+              for (part in moduleName.split(".")) {
+                new { value = part }
+              }
+            }
+          }
+          moduleExtendsOrAmendsClause {
+            type = "extends"
+            extendedModule = "package://pkg.pkl-lang.org/pkl-k8s/k8s@1.0.1#/K8sResource.pkl"
+          }
+        }
+      }
+      properties {
+        new {
+          modifiers { "fixed" }
+          name { value = "apiVersion" }
+          typeAnnotation {
+            type = new TypeNode.StringLiteralTypeNode {
+              value = "\(crd.spec.group)/\(version.name)"
+            }
+          }
+        }
+        new {
+          modifiers { "fixed" }
+          name { value = "kind" }
+          typeAnnotation {
+            type = new TypeNode.StringLiteralTypeNode {
+              value = crd.spec.names.kind
+            }
+          }
+        }
+        new {
+          name { value = "metadata" }
+          docComment {
+            value = """
+              Standard object's metadata.
+
+              More info: <https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata>
+              """
+          }
+          typeAnnotation {
+            type = new TypeNode.NullableTypeNode {
+              typeNode = new TypeNode.DeclaredTypeNode {
+                name {
+                  parts {
+                    new {
+                      value = "ObjectMeta"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+        when (isClassLike(filteredRootSchema)) {
+          ...generateClassBody(filteredRootSchema, allTypeNames)
+        }
+      }
+
+    }
+
+/// Determine the doc comments of a schema.
+///
+/// This combines a schema's title, description and "default" descriptions into one doc comment.
+/// Wraps the comments at 100 columns.
+local function getDocComment(schema: JsonSchema|Boolean, type: "class"|"module"|Null): DocCommentNode? =
+  if (schema is Boolean)
+    null
+  else
+    let (docCommentText: String =
+      List(
+        schema.title,
```

Comment:
It's a little bit of a faux-pas to start the doccomment with a repeat of the thing (it's a stutter)
```suggestion
```